docs/content/sips/xxx-spin-plugins.md

```diff
@@ -0,0 +1,154 @@
+title = "SIP xxx - Plugin System for Spin"
+template = "main"
+date = "2022-08-09T13:22:30Z"
+---
+
+Summary: Plugin system for spin.
+
+Owner: karthik.ganeshram@fermyon.com
+
+Created: August 9, 2022 
+
+## Background 
+
+As the functionality fo spin gets extended, there will be a point where every feature will not be required/used by every user. An example of this would be different triggers for spin along with possible subcommands to add more functionality to spin. Therefore it would make sense to be able to add features as plugins allowing for addition of only the required features witout modifying the spin binary.
+
+## Proposal
+
+Create a new subcommand for spin called `spin plugin` which will have two further subcommands namely
+
+- `install`
+- `uninstall`
+
+### Types of plugins
+
+![Classification of types of plugins](https://i.imgur.com/Fo7EGPQ.png)
+
+- Standalone Binary Plugins (external subcommands) - These work to add features to spin and do not need any use of the spin binary.
+- Internal Spin Plugins - These are plugins like triggers that are used by the spin application. 
+- Spin dependant User Plugins - These user invoked plugins require some functionality provided by spin.
+
+### Proposed workflow
+
+A plugin can be installed using the following command.
+
+```bash
+$ spin plugin install <name_of_plugin>
```

Comment:
The plugins can be directly placed in the install location for testing but a flag can be added to give these options.

Reply:
"Place it directly in the install location" isn't going to be great for a production environment though.  An operator is going to want to be able to script creation of the environment, and have confidence that the script is repeatable.

Reply:
That makes sense. Adding an option to install from a local file makes sense.

Reply:
This could be simply pointing to a local plugin-manifest.json rather than grabbing from remote, right? `spin plugin install some-plugin -f /path/to/some-plugin.json`. Installing a specific version could work similarly, where they always pass in the plugin-manifest with the version they want.

Reply:
There are two different things you could do here:
- pointing to a local manifest, which will lead to the actual binary being downloaded from one of the URLs in the manifest
- pointing to a local binary, where you just want to copy/symlink that binary into the plugins directory

Not sure which of these would be better for production install